### PR TITLE
Add more Examples

### PR DESCRIPTION
**Pull request by lukewilliamboswell**
This PR adds a number of new examples.

For popular problems updated references to refer to Wikipedia problem description.

### CODE REVIEW

**Comment by Anton-4** [comment]
Thanks @lukewilliamboswell! I've partly reviewed this and will continue tomorrow.

File: examples/AplusB/main.roc
Comment (Anton-4) [inline examples/AplusB/main.roc:1]: I think the chance is small that someone will look for an a + b example but we do demonstrate using args and writing to stout here. What do you think of making this a minimal advent of code template? We could also add search tags like project euler, programming contest, hackerrank...

File: examples/Arithmetic/main.roc [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+        bResult = List.get args 2 |> Result.try Str.toI32
+
+        when (aResult, bResult) is
+            (Ok a, Ok b) ->
+                if a < -1000 || a > 1000 || b < -1000 || b > 1000 then
+                    Task.fail InvalidNumStr
+                else
+                    Task.succeed { a, b }
+
+            _ -> Task.fail InvalidNumStr
```
Comment (Anton-4) [inline examples/Arithmetic/main.roc:65]: The `readArgs` function adds a lot of noise to a code example. I think we should stick to using expect all the time, except for examples like  "command line args", "advent of code", ...

File: examples/Arithmetic/main.roc [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+        results =
+            [
+                ("sum", args.a + args.b),
+                ("difference", args.a - args.b),
+                ("product", args.a * args.b),
+                ("integer quotient", args.a // args.b),
+                ("remainder", args.a % args.b),
+                ("exponentiation", Num.powInt args.a args.b),
+            ]
+            |> List.map calculate
```
Comment (Anton-4) [inline examples/Arithmetic/main.roc:30]: I think we should split out the vital part of the solution into a separate function. Readers are likely to just want to copy that function.

File: examples/Arithmetic/main.roc [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+TaskErrors : [InvalidArg, InvalidNumStr]
+
+main =
+    task =
+        args <- readArgs |> await
+
+        calculate = \(operation, result) ->
+            resultStr = Num.toStr result
+
+            "\(operation): \(resultStr)"
```
Comment (Anton-4) [inline examples/Arithmetic/main.roc:19]: Given that this function does no calculation, we should probably call it something else :p

File: examples/TowersOfHanoi/main.roc [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+        numDisksResult = List.get args 1 |> Result.try Str.toU32
+
+        when numDisksResult is
+            Ok numDisks ->
+                if numDisks < 1 then
+                    Task.fail InvalidNumStr
+                else
+                    Task.succeed { numDisks }
+
+            _ -> Task.fail InvalidNumStr
```
Comment (Anton-4) [inline examples/TowersOfHanoi/main.roc:49]: I would again remove the `readArgs` here, like mentioned earlier.

**Review by Anton-4 — CHANGES_REQUESTED** [review]

File: examples/AplusB/main.roc
Comment (lukewilliamboswell) [reply examples/AplusB/main.roc:1]: Ok, killed A+B and made it a HelloWorld instead

**Comment by lukewilliamboswell** [comment]
Thank you for the feedback @Anton-4. I agree we can keep them smaller and use `interface` modules to keep it simple. However, I would like to leave those examples as is, and pull them apart in a follow up PR. I would also like to get some kind of Tags or Search feature in, or at least think about the design, as these examples are going to grow wild pretty quickly if we have too many in just a big list.

**Review by Anton-4 — APPROVED** [review]
Sounds good! I like the diverse set of examples for pattern matching :)

**Pull request merged**